orchagent/neighorch.h

```diff
@@ -7,42 +7,83 @@
 #include "intfsorch.h"
 
 #include "ipaddress.h"
+#include "tokenize.h"
 
 #define NHFLAGS_IFDOWN                  0x1 // nexthop's outbound i/f is down
 
-struct NeighborEntry
+#define VRF_PREFIX "Vrf"
+extern IntfsOrch *gIntfsOrch;
+
+struct NextHopKey
 {
     IpAddress           ip_address;     // neighbor IP address
     string              alias;          // incoming interface alias
 
-    bool operator<(const NeighborEntry &o) const
+    NextHopKey() = default;
+    NextHopKey(const std::string &ipstr, const std::string &alias) : ip_address(ipstr), alias(alias) {}
+    NextHopKey(const IpAddress &ip, const std::string &alias) : ip_address(ip), alias(alias) {}
+    NextHopKey(const std::string &str)
+    {
+        if (str.find(',') != string::npos)
```

Comment:
Why do you need this check?

Reply:
It originally come from `ipAddress` class. Some apps use multi-ipaddresses string to create `NexthopKey` class. It needs to throw an exception to reminder the apps to use `NexthopGroupKey` class